includes/class-wp-job-manager-recaptcha.php

```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * File containing the WP_Job_Manager_Recaptcha class.
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WP_Job_Manager_Recaptcha class.
+ *
+ * @since $$next-version$$
+ * @internal
+ */
+class WP_Job_Manager_Recaptcha {
+
+	/**
+	 * Stores static instance of class.
+	 *
+	 * @var WP_Job_Manager_Recaptcha The single instance of the class.
+	 */
+	private static $instance = null;
+
+	/**
+	 * Site key.
+	 *
+	 * @var string
+	 */
+	private $site_key;
+
+	/**
+	 * Secret key.
+	 *
+	 * @var string
+	 */
+	private $secret_key;
+
+	/**
+	 * The reCAPTCHA version.
+	 *
+	 * @var string
+	 */
+	private $recaptcha_version;
+
+	/**
+	 * Initialize class for landing pages.
+	 *
+	 * @since 2.0.0
+	 */
+	public function __construct() {
+		$this->site_key          = get_option( 'job_manager_recaptcha_site_key' );
+		$this->secret_key        = get_option( 'job_manager_recaptcha_secret_key' );
+		$this->recaptcha_version = get_option( 'job_manager_recaptcha_version' );
+
+		if ( $this->use_recaptcha_field() ) {
+			add_action( 'submit_job_form_end', [ $this, 'display_recaptcha_field' ] );
+			add_filter( 'submit_job_form_validate_fields', [ $this, 'validate_recaptcha_field' ] );
+			add_filter( 'submit_draft_job_form_validate_fields', [ $this, 'validate_recaptcha_field' ] );
+		}
+	}
+
+	/**
+	 * Returns static instance of class.
+	 *
+	 * @return self Main instance.
+	 */
+	public static function instance() {
```

Comment:
We have a Singleton trait which you can use here I think

Reply:
Done in ed453f473f4f0b602a3117436c1666349d94a36f